src/modules/m_cloaking.cpp

```diff
@@ -309,15 +310,16 @@ class ModuleCloaking : public Module
 	Version GetVersion() CXX11_OVERRIDE
 	{
 		std::string testcloak = "broken";
+		irc::sockets::sockaddrs sa;
 		if (Hash)
 		{
 			switch (mode)
 			{
 				case MODE_HALF_CLOAK:
-					testcloak = prefix + SegmentCloak("*", 3, 8) + suffix;
+					testcloak = GenCloak(sa, "", testcloak + std::to_string(domainparts));
```

Comment:
Yes, it breaks. But I'm not sure what's the alternative to sync `domainparts` then. I can only think about some complicated way using `OnSyncNetwork` which I would like to avoid.